### ISSUE TITLE
Integrate react-native inside of mobile apps

### ISSUE THREAD

**Issue by hellais** [ooni/probe#868]
Since we have some react skills in the OONI team it may be beneficial to start implementing some of the mobile apps views using react-native and consider eventually replacing all the UIs with react-native UIs.

This can be done incrementally starting from certain views only. 

The guide to follow would be the one on the react website for [Integration with existing . apps](https://facebook.github.io/react-native/docs/integration-with-existing-apps).

Things to consider in doing this are:

* How big will the app size increase?
* How much code can we share and re-use across Android and iOS?
* What other limitations are there in this approach?

**Comment by bassosimone** [comment]
@hellais this is clearly an epic! :-)

**Comment by lorenzoPrimi** [comment]
Maybe the news feed could be done in react 
https://github.com/ooni/probe/issues/837
<img width="762" alt="Screenshot 2019-07-14 at 12 01 43" src="https://user-images.githubusercontent.com/1616097/61182112-3555cd00-a62f-11e9-87e0-d076532cc9c2.png">

**Comment by lorenzoPrimi** [comment]
The android apk without react native is 22.9 MB, the aab is 22.4 mb. 
The aab shipped to the store will occupy between 13.1 and 13.8 MB on device
The apk with react-native is 46M @sarathms 

The iOS app with react native is 21.9 MB and without is 19.8 MB

**Comment by hellais** [comment]
> The android apk without react native is 22.9 MB
> The apk with react-native is 46M

Is this a comparison done using the same exact build or is it using a different build process?

IIRC @sarathms mentioned he was making a development build and that it wasn't good for comparison.

**Comment by lorenzoPrimi** [comment]
Pull Requests:
https://github.com/ooni/probe-android/pull/266
https://github.com/ooni/probe-ios/pull/313

**Comment by lorenzoPrimi** [comment]
@sarathms can I know which command you used to generate an apk (or aab)
when I generate it with Android Studio it doesn't start the webserver

**Comment by lorenzoPrimi** [comment]
My builds:
full .aab 42.3 MB (actual size on device 18.3 to 19.8 MB)
full .apk 42.8 MB
fdroid .apk 41.8 MB

**Comment by lorenzoPrimi** [comment]
Generating different apk for every atchitecture reduces the size, like stated in 
https://facebook.github.io/react-native/docs/signed-apk-android

19M  app-fdroid-arm64-v8a-release.apk
19M app-fdroid-armeabi-v7a-release.apk
19M app-fdroid-x86-release.apk
20M app-fdroid-x86_64-release.apk

**Comment by lorenzoPrimi** [comment]
React native builds:
Full .aab 42.3 MB (actual size on device 18.3 to 19.8 MB)
Full .apk 42.8 MB
Fdroid .apk 41.8 MB

Old versions:
Full and fdroid apk is 22.9 MB
Full aab 22.4MB (13.1 and 13.8 MB on device)

So the Fdroid. apk is 20MB bigger, I tried many solutions, using proguard or hermes and I never got it smaller than 40 MB.

According to the official website https://facebook.github.io/react-native/docs/signed-apk-android
this happens because in the normal apk there will be packed fdroid builds for every possible arch.

The way to go is to use only the new aab file which only install the required arch on the relative phone, but this wouldn't work on fdroid.

A way I found to have a decent size apk is to generate a different apk for every arch, in this case we are aligned with the original apk size, even better. We need to understand if this would work for fdroid release or the want the full apk.

```
19M app-fdroid-arm64-v8a-release.apk
19M app-fdroid-armeabi-v7a-release.apk
19M app-fdroid-x86-release.apk
20M app-fdroid-x86_64-release.apk
```
